### PR TITLE
fix: Fixed swagger error. Fixes #8922

### LINKED ISSUE #8922: Have errors for swagger.json when opening with https://editor.swagger.io
I found there are some errors when opening _.primary.swagger.json with https://editor.swagger.io/ like below:
image

Errors
Semantic error at paths./artifact-files/{namespace}/{idDiscriminator}/{id}/{nodeId}/{artifactDiscriminator}/{artifactName}.get.responses.default.schema.$ref
$refs must reference a valid location in the document
Jump to line 71
Semantic error at paths./artifacts/{namespace}/{name}/{nodeId}/{artifactName}.get.responses.default.schema.$ref
$refs must reference a valid location in the document
Jump to line 104
Structural error at paths./input-artifacts/{namespace}/{name}/{nodeId}/{artifactName}.get.responses.200
should NOT have additional properties additionalProperty: content
Jump to line 129
Semantic error at paths./input-artifacts/{namespace}/{name}/{nodeId}/{artifactName}.get.responses.default.schema.$ref
$refs must reference a valid location in the document
Jump to line 138
Structural error at paths./artifacts-by-uid/{uid}/{nodeId}/{artifactName}.get.responses.200
should NOT have additional properties additionalProperty: content
Jump to line 159
Semantic error at paths./artifacts-by-uid/{uid}/{nodeId}/{artifactName}.get.responses.default.schema.$ref
$refs must reference a valid location in the document
Jump to line 168
Semantic error at paths./input-artifacts-by-uid/{uid}/{nodeId}/{artifactName}.get.responses.default.schema.$ref
$refs must reference a valid location in the document
Jump to line 197
Semantic error at definitions.github.com.argoproj.argo_workflows.v3.pkg.apis.workflow.v1alpha1.ParallelSteps.items.$ref
$refs must reference a valid location in the document
Jump to line 206

### PR DESCRIPTION
**Pull request by JessieTeng89**
Signed-off-by: Teng, Jessie <yilin.teng@fmr.com>

fix: fix swagger error. Fixes https://github.com/argoproj/argo-workflows/issues/8922

### CODE REVIEW

**Comment by JessieTeng89** [comment]
Hi @alexec 
I created  a PR to fix #8922, kindly help to review it if you are free, Thanks

File: pkg/apiclient/_.primary.swagger.json [hunk 1]
```diff
@@ -301,6 +297,38 @@
...
+            "$ref": "#/definitions/google.protobuf.Any"
+          }
+        },
+        "error": {
+          "type": "string"
+        },
+        "message": {
+          "type": "string"
+        }
+      }
```
Comment (tczhao) [inline pkg/apiclient/_.primary.swagger.json:331]: overall looks good,
I'm not sure if we need to redefine these here though.
These are defined in api/openapi-spec/swagger.json

Comment (JessieTeng89) [reply pkg/apiclient/_.primary.swagger.json:331]: Yes, we got the definitions not found error when opening pkg/apiclient/_.primary.swagger.json, so I copied them from the file you mentioned

Comment (tczhao) [reply pkg/apiclient/_.primary.swagger.json:331]: I think the _.primary.swagger.json is an auxiliary file that helps to construct api/openapi-spec/swagger.json
We should just validate api/openapi-spec/swagger.json

Comment (JessieTeng89) [reply pkg/apiclient/_.primary.swagger.json:331]: Well, I only checked the swagger file for pkg/apiclient/_.primary.swagger.json before as the related issue is only to fix the error in it.
And I tried to open api/openapi-spec/swagger.json just now, there is no error.
Can you help to confirm if this PR is still needed or we can close the the issue and PR?

Comment (JessieTeng89) [reply pkg/apiclient/_.primary.swagger.json:331]: @tczhao I saw this pull request is already approved, can we merge it or need some other actions from my side, kindly let me know if needed

Comment (JessieTeng89) [reply pkg/apiclient/_.primary.swagger.json:331]: Hi @tczhao @alexec 
A gentle follow up with this PR, I am looking forward for your feedback 😄

**Review by tczhao — APPROVED** [review]

**Comment by JessieTeng89** [comment]
@tczhao Thanks for your approval, can we merge this to master branch or need to ask someone else to do it?

**Pull request merged**